LoopStructural/interpolators/_geological_interpolator.py

```diff
@@ -125,7 +125,7 @@ def set_gradient_constraints(self, points: np.ndarray):
         -------
 
         """
-        if points.shape[1] < 7:
+        if points.shape[1] < 6:
```

Comment:
I would rather keep the size of the array to include weights but to add a column if the array doesn't include weights. 
```
if points.shape[1] == 6:
   points = np.hstack([points,np.ones(points.shape[0],1)])
```
Because locally weighting constraints is something that should be possible.